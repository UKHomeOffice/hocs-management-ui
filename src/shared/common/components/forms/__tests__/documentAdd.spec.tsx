import React from 'react';
import { render, screen } from '@testing-library/react';
import DocumentAdd from '../documentAdd';
import '@testing-library/jest-dom';

const FIELD_NAME = 'add-document';

describe('Document add component', () => {

    it('should render with default props', () => {
        const wrapper = render(<DocumentAdd name={FIELD_NAME} updateState={() => null} />);
        expect(wrapper).toBeDefined();
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with correct label when passed', () => {
        const label = 'MY_LABEL';
        render(<DocumentAdd name={FIELD_NAME} updateState={() => null} label={label} />);

        expect(screen.getByText('MY_LABEL')).toBeInTheDocument();
    });

    it('should render with validation error when passed', () => {
        const error = 'MY_ERROR';
        render(<DocumentAdd name={FIELD_NAME} updateState={() => null} error={error} />);
        expect(screen.getByText('MY_ERROR')).toBeInTheDocument();
    });

    it('should render with hint when passed', () => {
        const hint = 'MY_HINT';
        render(<DocumentAdd name={FIELD_NAME} updateState={() => null} hint={hint} />);
        expect(screen.getByText('MY_HINT')).toBeInTheDocument();
    });
});
