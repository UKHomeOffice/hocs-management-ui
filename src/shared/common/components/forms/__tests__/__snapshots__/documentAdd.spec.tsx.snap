// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document add component should render with correct label when passed 1`] = `
<DocumentAdd
  label="MY_LABEL"
  name="add-document"
  updateState={[Function]}
>
  <div
    className="govuk-form-group"
  >
    <label
      className="govuk-label"
      htmlFor="add-document"
      id="add-document-label"
    >
      <label
        className="govuk-label govuk-label--s"
        htmlFor="add-document"
        id="add-document-label"
      >
        MY_LABEL
      </label>
    </label>
    <input
      className="govuk-file-upload"
      id="add-document"
      name="add-document"
      onChange={[Function]}
      type="file"
    />
  </div>
</DocumentAdd>
`;

exports[`Document add component should render with default props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="govuk-form-group"
      >
        <label
          class="govuk-label"
          for="add-document"
          id="add-document-label"
        >
          <label
            class="govuk-label govuk-label--s"
            for="add-document"
            id="add-document-label"
          >
            Add document
          </label>
        </label>
        <input
          class="govuk-file-upload"
          id="add-document"
          name="add-document"
          type="file"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="govuk-form-group"
    >
      <label
        class="govuk-label"
        for="add-document"
        id="add-document-label"
      >
        <label
          class="govuk-label govuk-label--s"
          for="add-document"
          id="add-document-label"
        >
          Add document
        </label>
      </label>
      <input
        class="govuk-file-upload"
        id="add-document"
        name="add-document"
        type="file"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Document add component should render with hint when passed 1`] = `
<DocumentAdd
  hint="MY_HINT"
  name="add-document"
  updateState={[Function]}
>
  <div
    className="govuk-form-group"
  >
    <label
      className="govuk-label"
      htmlFor="add-document"
      id="add-document-label"
    >
      <label
        className="govuk-label govuk-label--s"
        htmlFor="add-document"
        id="add-document-label"
      >
        Add document
      </label>
      <span
        className="govuk-hint"
      >
        MY_HINT
      </span>
    </label>
    <input
      className="govuk-file-upload"
      id="add-document"
      name="add-document"
      onChange={[Function]}
      type="file"
    />
  </div>
</DocumentAdd>
`;

exports[`Document add component should render with validation error when passed 1`] = `
<DocumentAdd
  error="MY_ERROR"
  name="add-document"
  updateState={[Function]}
>
  <div
    className="govuk-form-group"
  >
    <label
      className="govuk-label"
      htmlFor="add-document"
      id="add-document-label"
    >
      <label
        className="govuk-label govuk-label--s"
        htmlFor="add-document"
        id="add-document-label"
      >
        Add document
      </label>
      <span
        className="govuk-error-message"
        id="add-document-error"
      >
        MY_ERROR
      </span>
    </label>
    <input
      className="govuk-file-upload"
      id="add-document"
      name="add-document"
      onChange={[Function]}
      type="file"
    />
  </div>
</DocumentAdd>
`;
